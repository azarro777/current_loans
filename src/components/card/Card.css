.card {
	position: relative;
	display: flex;
	justify-content: center;
	align-items: center;
	width: 676px;
	height: 134px;
	border: 1px solid rgba(48, 50, 71, 0.1);
	border-radius: 20px;
	margin: 30px 64px;
}

.card-container {
	
}

.card-wraper {
	width: 612px;
	height: 70px;
	display: flex;
	justify-content: space-between;
}

.card-title {
	width: 151px;
	height: 30px;
	overflow: hidden;
	padding: 0;
	margin: 0;
	font-family: 'Inter';
	font-style: normal;
	font-weight: 600;
	font-size: 25px;
	line-height: 30px;
	color: #303247;
}

.card-text {
	padding: 0;
	margin-top: 15px;
	font-family: 'Inter';
	font-style: normal;
	font-weight: 400;
	font-size: 20px;
	line-height: 24px;
	color: #303247;
}

.card-invested {
	position: absolute;
	width: 67px;
	height: 20px;
	left: 577px;
	top: -5px;

	font-family: 'Inter';
	font-style: normal;
	font-weight: 600;
	font-size: 16px;
	line-height: 20px;

	color: #51A571;
}

.card-button {
	margin-top: auto;
	width: 137px;
	height: 56px;
	font-family: 'Inter';
	font-style: normal;
	font-weight: 500;
	font-size: 20px;
	line-height: 24px;
	text-transform: uppercase;
	color: #303247;
	background: #F7E308;
	border: none;
	border-radius: 5px;
}