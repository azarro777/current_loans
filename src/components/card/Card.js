import { useDispatch, useSelector } from 'react-redux';
import { setModal, setModalData } from '../../reducers/appReducer';
import './Card.css';

export const Card = (props) => {
	const {id, title, tranche, available, term_remaining} = props;
	const loans = useSelector(state => state.app.loans);
	const dispatch = useDispatch();

	const modalHandler = () => {
		dispatch(setModal(true));
		dispatch(setModalData({id, available, term_remaining}));
		console.log("Modal", props); //! Console log!
		console.log("card", id, loans.find(item => item.id === id)); //! Console log!
	};

	return (
		<div className="card">
			<div className="card-wraper">
				<div className="card-container">
					<h2 className="card-title">{title}</h2>
					<h3 className="card-text">{tranche}</h3>
				</div>
				<button className="card-button" onClick={() => modalHandler()}>INVEST</button>
			</div>
		</div>
);
};