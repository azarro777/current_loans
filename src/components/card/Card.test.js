import React from 'react';
import { Card } from './Card';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import thunk from 'redux-thunk';

describe("on mount", () => {
  it("dispatch search action to store", () => {
		const mockStore = configureMockStore([thunk])([]);;
		const store = mockStore;
		shallow(<Provider store={store}><Card/></Provider>)
    const actions = store.getActions();
    expect(actions).toEqual([]);
  });
	it('Card have rendered correctly', () => {
		const mockStore = configureMockStore([thunk])([]);;
		const store = mockStore;
		const card = shallow(<Provider store={store}><Card/></Provider>);
		expect(card).toMatchSnapshot();
	});
});
