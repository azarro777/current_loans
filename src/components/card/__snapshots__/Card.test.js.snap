// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`on mount Card have rendered correctly 1`] = `ShallowWrapper {}`;
